components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.websocket/src/main/java/org/wso2/carbon/event/output/adapter/websocket/WebsocketEventAdapter.java

```diff
@@ -121,26 +122,35 @@ public void connect() {
         try {
             session = client.connectToServer(new WebsocketClient(), clientEndpointConfig, new URI(socketServerUrl));
         } catch (DeploymentException e) {
-            throw new ConnectionUnavailableException("The adaptor " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
-                    socketServerUrl, e);
+            throw new ConnectionUnavailableException("The adapter " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
+                                                     socketServerUrl, e);
         } catch (IOException e) {
-            throw new ConnectionUnavailableException("The adaptor " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
-                    socketServerUrl, e);
+            throw new ConnectionUnavailableException("The adapter " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
+                                                     socketServerUrl, e);
         } catch (URISyntaxException e) {
-            throw new OutputEventAdapterRuntimeException("The adaptor " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
-                    socketServerUrl, e);
+            throw new OutputEventAdapterRuntimeException("The adapter " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
+                                                         socketServerUrl, e);
         }
     }
 
     @Override
     public void publish(Object message, Map<String, String> dynamicProperties) {
-
         try {
-            executorService.execute(new WebSocketSender(message.toString()));
+            if (session == null) {
+                throw new ConnectionUnavailableException("Connection was not available when trying to publish a message via the adapter: "
+                                                         + eventAdapterConfiguration.getName() + ", for tenant ID: " + tenantId
+                                                         + " as the session was null. \n Message: \n" + message);
```

Comment:
@dilini-muthumala 
Dilini, can we make the error message a bit shorter and a bit more explanatory for the user? We can provide the failing event adapter/publisher name and tenant id + message. But since it is throwing a 'ConnectionUnavailableException', including the complete sentence might be a bit redundant and cause the log lines to grow IMHO. Also, I think it will be more explanatory to the user if the log says 'no session is available' instead of session is 'null'. Additionally, since we are using the term 'event' all across CEP adapters, I think better to use the term 'Event' instead of 'Message' for consistency. WDYT?